### PR TITLE
Add `contrastLevel` parameter to `ColorScheme.fromSeed`

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR is to add a parameter `contrastLevel` to `ColorScheme.fromSeed` so that we can construct high contrast `ColorScheme`.

https://github.com/flutter/flutter/assets/36861262/c609c996-5dfe-4c6c-800c-349a99de4256

Related to https://github.com/flutter/flutter/issues/149683

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/color_scheme.dart [hunk 1]
```diff
@@ -283,9 +283,14 @@ class ColorScheme with Diagnosticable {
   /// If the resulting color scheme is too dark, consider setting `dynamicSchemeVariant`
   /// to [DynamicSchemeVariant.fidelity], whose palettes match the seed color.
   ///
+  /// The `contrastLevel` parameter indicates the contrast level between color
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/color_scheme.dart:286]: suggest mentioning that medium and high contrast from the M3 guidelines correspond to 0.5 and 1.0 respectively.

**Review by guidezpl — APPROVED** [review]

File: packages/flutter/lib/src/material/color_scheme.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/color_scheme.dart:286]: Thanks for helping review this! Will update!

**Comment by QuncCccccc** [comment]
Once this is land, will add the high-contrast `ColorScheme` to the a11y-assessments app:) @chunhtai

File: examples/api/lib/material/color_scheme/color_scheme.0.dart [hunk 2]
```diff
@@ -10,6 +10,7 @@ const Widget divider = SizedBox(height: 10);
 
 void main() => runApp(const ColorSchemeExample());
 
+
```
Comment (chunhtai) [inline examples/api/lib/material/color_scheme/color_scheme.0.dart:13]: additional line?

**Review by chunhtai — COMMENTED** [review]
LGTM, is there another pr to update the a11y app as well?

**Review by chunhtai — APPROVED** [review]

**Comment by QuncCccccc** [comment]
> LGTM, is there another pr to update the a11y app as well?

Yes! Will create another PR to add the high-contrast color scheme to the a11y assessment app.

File: examples/api/lib/material/color_scheme/color_scheme.0.dart (hunk 2 above)
Comment (QuncCccccc) [reply examples/api/lib/material/color_scheme/color_scheme.0.dart]: Updated:) Thanks for the review!

**Pull request merged**